registry/quilt_server/views.py

```diff
@@ -503,14 +507,26 @@ def _mp_track(**kwargs):
     mp.track(distinct_id, MIXPANEL_EVENT, all_args)
 
 def _generate_presigned_url(method, owner, blob_hash):
-    return s3_client.generate_presigned_url(
+    url = s3_client.generate_presigned_url(
         method,
         Params=dict(
             Bucket=PACKAGE_BUCKET_NAME,
             Key='%s/%s/%s' % (OBJ_DIR, owner, blob_hash)
         ),
         ExpiresIn=PACKAGE_URL_EXPIRATION
     )
+    # This is similar to what's done in the google documentation for a simple
+    # migration, but for pre-signed URLs (for which no example is given).
+    #    https://cloud.google.com/storage/docs/migrating#keys
+    parsed_url = urlparse(url)
+    if parsed_url.hostname.lower() in GOOGLE_STORAGE_ENDPOINTS:
+        parsed_query = parse_qsl(parsed_url.query)
+        for index, item in enumerate(parsed_query):
```

Comment:
Hm, this looks unnecessarily difficult... Would something like this work? https://stackoverflow.com/questions/2506379/add-params-to-given-url-in-python I guess it won't preserve the order, but I can't imagine that making a difference.

Reply:
*shrug* this retains order, the other doesn't.  The complexity's about the same, though this has more lines.  I can shoot for brevity here.

Reply:
Fixed.